api-client/src/main/java/com/xing/api/data/messages/MessageAttachment.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2016 XING SE (http://xing.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.xing.api.data.messages
+
+import com.squareup.moshi.Json
+import com.xing.api.data.SafeCalendar
+import java.io.Serializable
+
+/**
+ * Java representation of an Attachment on the Message.
+ */
+data class MessageAttachment(
+        @Json(name = "id")
+        val id: String,
+
+        @Json(name = "created_at")
+        val createdAt: SafeCalendar,
+
+        @Json(name = "filename")
+        val fileName: String,
+
+        @Json(name = "mime_type")
+        val mimeType: String,
+
+        @Json(name = "size")
+        val fileSize: Int
+) : Serializable {
+
+    override fun equals(other: Any?):Boolean = other is MessageAttachment && id == other.id
```

Comment:
I don't think this overriding is necessary. If there are two different attachments with the same id it is an implementation error which should not be covered by a data class

Reply:
I just converted the existing class into Kotlin without modifying the original behaviour. IMO the "equals with ids or all fields" should be handled separately, otherwise we risk changing a behaviour in a simple conversion PR and breaking apps using the lib, no?

Reply:
Having the conversion to kotlin could already produce crashes - IllegalStateExceptions if initialising a value with null to name one.

So I think fixing a wrong implementation could also be part of it.

We can add it to the changelog to make it visible.